pkg/apic/oas2Swagger.go

```diff
@@ -38,9 +42,51 @@ func (o *oas2Swagger) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	}
 
 	// unmarshal new byte array
-	var newVal openapi2.Swagger
+	var newVal openapi2.T
 	yaml.Unmarshal(newBytes, &newVal)
 
-	o.Swagger = newVal
+	o.T = newVal
+	return nil
+}
+
+// ParseOAS2 converts a JSON spec into an OpenAPI2 object
+func ParseOAS2(spec []byte) (*openapi2.T, error) {
+	swaggerObj := &openapi2.T{}
+	err := json.Unmarshal(spec, swaggerObj)
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.Contains(swaggerObj.Swagger, "2.") {
+		return nil, errors.New(oasParseError("2.0", "'swagger' must be version '2.0'."))
+	}
+	if swaggerObj.Info.Title == "" {
+		return nil, errors.New(oasParseError("2.0", "'info.title' key not found."))
+	}
+	if swaggerObj.Paths == nil {
+		return nil, errors.New(oasParseError("2.0", "'paths' key not found."))
+	}
+	return swaggerObj, nil
+}
+
+// SetHostDetails Updates the Host, BasePath, and Schemes fields on an oas2Swagger object
+func SetHostDetails(spec *openapi2.T, endpointURL string) error {
```

Comment:
What would be an appropriate struct for it? I reverted the changes I had to expose the spec parsers in an attempt to keep those private, since I didn't see a need for them.

Reply:
I see the problem with yaml parsing and losing the key definitions that needs to be address separately. Can you for now change the name of SetHostDetails to SetOAS2HostDetails (or something neat) and SetServers to SetOAS3Servers ?